src/components/icon/icon.spec.js

```diff
@@ -0,0 +1,406 @@
+        type: 'services', theme: classicTheme, isFont: true, fontSize: 'large'
+      });
+      assertStyleMatch(
+        {
+          marginTop: '-8px'
+        },
+        wrapper.toJSON(),
+        { modifier: '&::before' }
+      );
+    });
+  });
+
+  describe('with custom class name', () => {
+    it('renders with a custom classname', () => {
+      const wrapper = render({ className: 'testClass' });
+      expect(wrapper.find('.testClass').length).toEqual(1);
+    });
+  });
+
+
+  describe('icon color', () => {
+    it('renders proper icon color for disabled state', () => {
+      const wrapper = renderStyles({ disabled: true, bgTheme: 'business' });
+      assertStyleMatch(
+        {
+          color: baseTheme.icon.disabled
+        },
+        wrapper.toJSON()
+      );
+    });
+
+    describe.each(['error', 'info', 'business'])(
+      'whent the background theme is %s',
+      (whiteIconBackground) => {
+        it('renders a white icon', () => {
+          const wrapper = renderStyles({ bgTheme: whiteIconBackground });
+          assertStyleMatch(
+            {
+              color: baseTheme.colors.white
+            },
+            wrapper.toJSON()
+          );
+        });
+      }
+    );
+
+    describe.each(['success', 'warning'])(
+      'when the background theme is %s',
+      (darkIconBackground) => {
+        it('renders a dark icon', () => {
+          const wrapper = renderStyles({ bgTheme: darkIconBackground });
+          assertStyleMatch(
+            {
+              color: baseTheme.icon.default
+            },
+            wrapper.toJSON()
+          );
+
+          assertStyleMatch(
+            {
+              color: baseTheme.icon.defaultHover
+            },
+            wrapper.toJSON(),
+            { modifier: ':hover' }
+          );
+        });
+      }
+    );
+
+    describe('when bgTheme is set to none', () => {
+      it('renders white icon when iconColor is set to onDarkBackground', () => {
+        const wrapper = renderStyles({ iconColor: 'on-dark-background', bgTheme: 'none' });
+        assertStyleMatch(
+          {
+            color: baseTheme.colors.white
+          },
+          wrapper.toJSON()
+        );
+      });
+
+      it('renders dark icon when iconColor is set to onLightBackground', () => {
+        const wrapper = renderStyles({ iconColor: 'on-light-background', bgTheme: 'none' });
+        assertStyleMatch(
+          {
+            color: baseTheme.icon.onLightBackground
+          },
+          wrapper.toJSON()
+        );
+
+        assertStyleMatch(
+          {
+            color: baseTheme.icon.onLightBackgroundHover
+          },
+          wrapper.toJSON(),
+          { modifier: ':hover' }
+        );
+      });
+
+      describe('fontSize is "large" and bgSize is "small"', () => {
+        it('adjusts bgSize to "medium"', () => {
+          const wrapper = renderStyles({ fontSize: 'large', bgTheme: 'foo', bgSize: 'small' });
+          assertStyleMatch(
```

Comment:
I'm not sure that the assertion works for this description here, please could you tweak one or the other.

Reply:
Yeah good point the description is certainly unclear, I'll update it now